incubator/hnc/internal/validators/namespace_test.go

```diff
@@ -12,29 +12,53 @@ import (
 )
 
 func TestDeleteSubNamespace(t *testing.T) {
-	// Create a namespace with owner annotation.
+	// Create a namespace instance a.
 	sub := &corev1.Namespace{}
 	sub.Name = "a"
-	setSubAnnotation(sub, "b")
 
-	// a (subnamespace of b) -> b
-	f := foresttest.Create("B-")
-	vns := &Namespace{Forest: f}
+	// We will put an anchor a in the namespace c if it exists.
+	tests := []struct {
+		name           string
+		forest         string
+		subnamespaceOf string
```

Comment:
Maybe comment this a bit more? E.g., "we'll use the forest to create only full namespaces, and then manually turn them into subnamespaces by adding the annotations" or something like that?